### PR TITLE
win32: opt in to the windows segment heap

### PR DESCRIPTION
**Pull request by Andarwinux**
ref: https://learn.microsoft.com/en-us/windows/win32/sbscs/application-manifests#heaptype

The speed of releasing the demuxer cache has been greatly improved, from more than 10 seconds down to 1 second. (tested on win11 22h2 with `--demuxer-max-bytes=16384MiB`)

### CODE REVIEW

**Review by sfan5 — APPROVED** [review]
if it works, why not

**Comment by sfan5** [comment]
```
19:58 <sfan5> any windows user want to test #13137 before I merge it?
19:59 <JEEB> ah so the segment heap speeds things up with the cache
20:01 <JEEB> I think the only thing I'd do is add a note to the commit message regarding the apparent speedup
20:01 <JEEB> which can be done when the commit is pulled in
20:02 <JEEB> funny how the manifest files reference doesn't say any negative possible effects for that thing
20:42 <llyyr> increases cpu usage apparently, which is why chrome reverted it on their end
20:43 <JEEB> right so it tries to optimize the chunks better
20:44 <sfan5> you mean this? https://chromium.googlesource.com/chromium/src/+/927879cc1fe389681fd43b5b4e91acc9bc1bf93d
20:49 <JEEB> https://bugs.chromium.org/p/chromium/issues/detail?id=1102281 so 5-6% to ~10% on web loads
20:49 <sfan5> reading through that they stopped pursuing the "how bad is segment heap really" question after switching to another different allocator
20:50 <JEEB> yea
20:50 <JEEB> partition alloc
20:50 <sfan5> and here <https://github.com/microsoft/Windows-Dev-Performance/issues/39#issuecomment-729313323> MS says they are working on making the old and new thing equal in performance
20:52 <JEEB> randomly searched the issues and there's something from 2022 https://github.com/microsoft/Windows-Dev-Performance/issues/106
20:54 <JEEB> that seems to mention "about 10% on average"
20:55 <sfan5> anyway it doesn't seem the overhead is too bad so I'd rather enable it and hope for the best
20:55 <sfan5> if someone wants to do more in-depth testing https://github.com/microsoft/mimalloc/ is worth a try
20:55 <sfan5> and whatever chromium uses (assuming that's portable)
20:58 <sfan5> "Were you able to route this issue to the right team?" lmao
20:58 <JEEB> yea I lol'd at this
20:58 <JEEB> *that
20:59 <sfan5> also yes it'd be good to include more context in the commit if we merge that
```

tl;dr please include some context in the commit message

**Comment by kasper93** [comment]
Should be good, little bit more overhead is expected, but unless someone find a real issue we can try to use it.
```
 legacy heap: 12639.917 ms
segment heap:  4389.989 ms
    mimalloc:  2465.386 ms
```

> Greatly improves demuxer cache

Sounds like it resolves the problem, while it only improves it. But anyway, it is ok. Still if you page out this memory, you are out of luck with any allocator, because most of the time is page faulting.

**Comment by kasper93** [comment]
> But is there any way to statically link mimalloc and override the standard malloc?

Not really, in practice.

> Maybe the mpv side could be easily modified to use mi_*, but that's not realistic for hundreds of dependencies.

Not worth probably to do that for mpv alone, there are a lot allocations on ffmpeg side too. In practice overriding malloc with dynamically linked mimalloc is easy and robust way to do it, but probably not worth it too in fact. And would be something to do in build bots if anything.
